ALICE3/Tasks/pidRICHqa.cxx

```diff
@@ -354,6 +382,11 @@ struct richPidQaMc {
         histos.fill(HIST("fRICH/pSec"), track.p());
         histos.fill(HIST("fRICH/ptSec"), track.pt());
       }
+      fillNsigmafRICH<0>(track, mcParticle, mcParticles, nsigma);
+      fillNsigmafRICH<1>(track, mcParticle, mcParticles, nsigma);
+      fillNsigmafRICH<2>(track, mcParticle, mcParticles, nsigma);
+      fillNsigmafRICH<3>(track, mcParticle, mcParticles, nsigma);
+      fillNsigmafRICH<4>(track, mcParticle, mcParticles, nsigma);
```

Comment:
Why do you call the function for all indices and check `pdgCode` each time inside?

Reply:
Because it's different at every call

Reply:
I mean, the first time I check the electron PDG, the second the muon and so on

Reply:
My point is that you call the function for every possible PDG and only inside the function checks whether it matches any value in the `PDGs` array and if it doesn't the function does nothing.
Why don't you first check the particle PDG and then call the function once with the correct index if found?

Reply:
Ah yes, I got your point, but I also need the index to identify the PDG and the name of the histogram that I want to fill.
To get what you suggest done in few lines I would probably need a loop and this cannot be done with the current implementation

Reply:
Or maybe a switch, but it would be more code anyways

Reply:
Something like this?
```c++
for (const uint8_t pidIndex : PDGs) {
  if (std::abs(particle.pdgCode()) == PDGs[pidIndex]) {
    fillNsigmafRICH<pidIndex>(track, mcParticle, mcParticles, nsigma);
    break;
  }
}
```

Reply:
eh but can it be passed to the template?

Reply:
it yes, this is indeed the best solution

Reply:
Not sure, but would be nice if it was.

Reply:
you can use the static for loop I added a while ago:
https://github.com/AliceO2Group/O2Physics/blob/master/Tutorials/src/histogramRegistry.cxx#L227-L236

Reply:
Hi, so going back on this, the solution proposed by Vit does not compile and the solution proposed by Mario is still a bit obscure for me. Maybe I would keep it like this, which seems the most simple to me

Reply:
just for completeness, this is what it would look like with the static for loop:
````
#include "Framework/StaticFor.h"
static_for<0, 4>([&](auto i) {  
  constexpr int pidIndex = i.value;
  fillNsigmafRICH<pidIndex>(track, mcParticle, mcParticles, nsigma);
});
````
this basicall expands the loop at compile time, avoiding having to do this manually